virl2_client/models/node.py

```diff
@@ -404,15 +405,48 @@ def boot_disk_size(self, value: int) -> None:
 
     @property
     def configuration(self) -> str | None:
-        """Return the initial configuration of this node."""
-        self.lab._sync_topology(exclude_configurations=False)
-        return self._configuration
+        """Return the contents of the main configuration file."""
+        self.lab.sync_topology_if_outdated(exclude_configurations=False)
+        return self._configuration[0] if self._configuration else None
 
     @configuration.setter
-    def configuration(self, value) -> None:
-        """Set the initial configuration of this node."""
+    def configuration(self, value: str | list | dict) -> None:
+        """
+        Set the content of:
+         - the main configuration file if passed a string,
+         - one configuration file if passed a dictionary in the format of
+        `{"name": "filename.txt", "content": "<file content>"}`,
+         - or multiple configuration files if passed a list of above dictionaries.
+        Can also use "Main" in place of the filename of the main configuration file.
+
+        :param value: The configuration data in one of three formats.
+        """
         self._set_node_property("configuration", value)
-        self._configuration = value
+        self._configuration = (
+            self._configuration if self._configuration is not None else []
+        )
+        if isinstance(value, str):
+            if not self._configuration:
+                self._configuration.append({"name": "Main", "content": value})
```

Comment:
shouldn't we get the data from BE instead?  I know that the patch node endpoint returns just node ID, but that can be changed

Reply:
I mean, possible, but then we should probably do that for other properties - not just configs - as to not introduce confusing inconsistency, so I'd rather just force a resync.